File: afc_saaspro_tax/afc_rest_apis/csharp/Client/AfcRestException.cs
using System;
using System.Net;

namespace Avalara.Comms.Rest.Sample.Client
{
    /// <summary>
    /// Exception class for errors thrown by the AfcRestClient.
    /// </summary>
    public class AfcRestException: Exception
    {
        /// <summary>
        /// HTTP status code returned by call to AFC REST service.
        /// </summary>
        public HttpStatusCode HttpStatusCode { get; protected set; }

        /// <summary>
        /// Constructor.
        /// </summary>
        /// <param name="httpCode">HTTP status code.</param>
        /// <param name="message">Error message.</param>
        /// <param name="innerException">Inner exception.</param>
        public AfcRestException(HttpStatusCode httpCode, string message, Exception innerException = null) :
            base(string.IsNullOrWhiteSpace(message) ? $"Request returned an error code: {httpCode}" : message, innerException)
        {
            HttpStatusCode = httpCode;
        }
    }
}

File: afc_saaspro_tax/afc_rest_apis/java/src/avalara/comms/rest/sample/client/AfcRestException.java
package avalara.comms.rest.sample.client;

/**
 * HTTP status code returned by call to AFC REST service.
 */
public class AfcRestException extends Exception {
    private int httpCode;
    
    /**
     * HTTP status code returned by call to AFC REST service.
     * @return HTTP status code.
     */
    public int getHttpCode() {
        return httpCode;
    }
    
    /**
     * Constructor.
     * 
     * @param httpCode  HTTP status code.
     * @param message   Exception message.
     */
    public AfcRestException(int httpCode, String message) {
        this(httpCode, message, null);
    }
    
    /**
     * Constructor.
     * 
     * @param httpCode          HTTP status code.
     * @param message           Exception message.
     * @param innerException    Inner exception.
     */
    public AfcRestException(int httpCode, String message, Exception innerException) {
        super(
            message == null || message.trim().length() == 0 ?
                "Request returned an error code: " + httpCode :
                message,
            innerException);
        this.httpCode = httpCode;
    }
}